ext/hook/uhook.c

```diff
@@ -789,6 +850,57 @@ ZEND_METHOD(DDTrace_HookData, overrideReturnValue) {
     RETURN_TRUE;
 }
 
+ZEND_METHOD(DDTrace_HookData, suppressCall) {
+    dd_hook_data *hookData = (dd_hook_data *)Z_OBJ_P(ZEND_THIS);
+
+    if (zend_parse_parameters_none() == FAILURE) {
+        return;
+    }
+
+    hookData->suppress_call = true;
```

Comment:
I'm fairly certain this approach doesn't work with inlining in JIT.

You may have to unconditionally blacklist hooks from which you wish to use call suppressions from being inlined: search for ZAI_JIT_BLACKLIST_ACTIVE in this file.

I.e. you will have to make this function accept boolean, always call it and just suppress when actually true, but unconditionally call the blacklisting.